src/java/com/zimbra/oauth/utilities/OAuth2DataSource.java

```diff
@@ -185,9 +186,9 @@ public void syncDatasource(ZMailbox mailbox, OAuthInfo credentials, Map <String,
                         "Missing import class for " + dsType.name() + " datasource type", null);
                 }
                 dsAttrs.put(Provisioning.A_zimbraDataSourceEnabled, "TRUE");
-                dsAttrs.put(Provisioning.A_zimbraDataSourceConnectionType, "cleartext");
+                dsAttrs.put(Provisioning.A_zimbraDataSourceConnectionType, "ssl");
                 dsAttrs.put(Provisioning.A_zimbraDataSourceOAuthRefreshToken, refreshToken);
-                dsAttrs.put(Provisioning.A_zimbraDataSourceHost, host);
+                dsAttrs.put(Provisioning.A_zimbraDataSourceHost, GoogleOAuth2Handler.GoogleCaldavConstants.HOST.getValue());
```

Comment:
we should not replace host here with Google endpoint, this is generic code for oauth. Replace this in the call from Google oauth handler

Reply:
This is the place where we are creating the datasource object and this host URL is used further.
Authentication is done by www.googleapis.com which is static url and have a different method to fetch token.

Reply:
are we sure that this method gets called in case of google authentication only? because name of the method signifies that it's generic method to sync token which can be called in case of any provider

Reply:
Changing host value changes are not required anymore.